provisioning/state_service.go

```diff
@@ -76,3 +87,65 @@ func EnsureMetal3StateService(info *ProvisioningInfo) (updated bool, err error)
 func DeleteMetal3StateService(info *ProvisioningInfo) error {
 	return client.IgnoreNotFound(info.Client.CoreV1().Services(info.Namespace).Delete(context.Background(), stateService, metav1.DeleteOptions{}))
 }
+
+// NewIronicServiceMonitor creates a ServiceMonitor for Ironic metrics
+func NewIronicServiceMonitor(namespace string) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": "monitoring.coreos.com/v1",
+			"kind":       "ServiceMonitor",
+			"metadata": map[string]interface{}{
+				"name":      ironicPrometheusExporterName,
+				"namespace": namespace,
+				"labels": map[string]interface{}{
+					cboLabelName: stateService,
+				},
+			},
+			"spec": map[string]interface{}{
+				"endpoints": []interface{}{
+					map[string]interface{}{
+						"port": metricsPortName,
+					},
+				},
+				"namespaceSelector": map[string]interface{}{
```

Comment:
(nit) namespaceSelector isn't strictly needed if the service monitor and pods live in the same namespace.

Reply:
dropped it